LibertyDSNP/example-client
# Retrieve Profiles from sdk (+ cleanup)

Purpose
---------------
This PR kind of got away from me. There's a lot of good stuff in here. I can't really pull it apart into separate PRs at this point, so it's going to be kind of tough to review. Here are the key points:

Solution
---------------

1. The example client now pulls all log history from the SDK. This is blocked by a [bug](https://github.com/LibertyDSNP/sdk-ts/pull/107) in the SDK. Until this is resolved, the PR is using a locally imported version of the SDK.
2. I've added a `scripts/populate.mjs` script to create locally hosted content on chain. It currently creates a profile and note message from each account.
3. I've modified the example client to use SDK types where possible. It has been using it's own version of activity content types in test data, which were either speculative or old. This change allows us to use data from the SDK directly. This is responsible for most of the churn in the PR. Note: a related rename of `src/utilties/activityPub.ts` -> `src/utilties/activityContent.ts` is being tracked as deletion and creation rather than a rename in git.
4. We now handle ActivityContentProfile announcements from the chain. These get stored in the profileSlice in redux. I've changed the structure of the profile slice to be a map keyed by dsnpId (aka socialAddress).
5. In `PostList.txs` we now match FeedItems to their associated Profiles and use that to substitute the user's name for their dsnpId. This is a bit of a proof-of-concept that we can associate this data. The actual UX may need to change.
6. In the process of fixing the tests for this PR, I addressed multiple warnings that were unrelated, but polluting the test output.

Steps to Verify
----------------
1. Start with a clean chain with deployed contracts.
2. run `node scripts/populate.mjs  --experimental-vm-modules` to create some content.
3. `npm run start`.
4. Click `All posts`.
5. You should see ~20 posts with names instead of ids.
6. `npm run test`
7. Verify test output is clean.
